src/main/scala/io/druid/indexer/spark/SparkDruidIndexer.scala

```diff
@@ -424,18 +423,18 @@ object SparkDruidIndexer {
 
 object SerializedJsonStatic {
   val LOG = new Logger("io.druid.indexer.spark.SerializedJsonStatic")
+  val defaultService = "spark-indexer"
+  // default indexing service port
+  val defaultPort = "8090"
   lazy val injector: Injector = {
     try {
       Initialization.makeInjectorWithModules(
         GuiceInjectors.makeStartupInjector(), List[Module](
           new Module {
             override def configure(binder: Binder): Unit = {
-              JsonConfigProvider
-                .bindInstance(
-                  binder,
-                  Key.get(classOf[DruidNode], classOf[Self]),
-                  new DruidNode("spark-indexer", null, null)
-                )
+              binder.bindConstant().annotatedWith(Names.named("serviceName")).to(defaultService)
```

Comment:
Can you just `@JacksonInject` instead of having to create a private injector here? as in, let Jackson do the injection in the original class?

Then you don't have to worry about any of this stuff or making it configurable.